test-data/unit/check-varargs.test

```diff
@@ -760,3 +760,232 @@ bar(*good3)
+    name: str
+    age: int
+
+class Base:
+    def foo(self, **kwargs: Unpack[Person]) -> None: ...
+class SubGood(Base):
+    def foo(self, *, name: str, age: int, extra: bool = False) -> None: ...
+class SubBad(Base):
+    def foo(self, *, name: str, age: str) -> None: ...  # E: Signature of "foo" incompatible with supertype "Base" \
+    # N:      Superclass: \
+    # N:          def foo(self, **kwargs: Unpack[Person]) -> None \
+    # N:      Subclass: \
+    # N:          def foo(self, *, name: str, age: str) -> None
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsGeneric]
+from typing import Generic, TypeVar
+from typing_extensions import Unpack, TypedDict
+
+T = TypeVar("T")
+class Person(TypedDict, Generic[T]):
+    name: str
+    value: T
+
+def foo(**kwargs: Unpack[Person[T]]) -> T: ...
+reveal_type(foo(name="test", value=42))  # N: Revealed type is "builtins.int"
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsInference]
+from typing import Generic, TypeVar, Protocol
+from typing_extensions import Unpack, TypedDict
+
+T_contra = TypeVar("T_contra", contravariant=True)
+class CBPerson(Protocol[T_contra]):
+    def __call__(self, **kwargs: Unpack[Person[T_contra]]) -> None: ...
+
+T = TypeVar("T")
+class Person(TypedDict, Generic[T]):
+    name: str
+    value: T
+
+def test(cb: CBPerson[T]) -> T: ...
+
+def foo(*, name: str, value: int) -> None: ...
+reveal_type(test(foo))  # N: Revealed type is "builtins.int"
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsOverload]
+from typing import Any, overload
+from typing_extensions import Unpack, TypedDict
+
+class Person(TypedDict):
+    name: str
+    age: int
+
+class Fruit(TypedDict):
+    sort: str
+    taste: int
+
+@overload
+def foo(**kwargs: Unpack[Person]) -> int: ...
+@overload
+def foo(**kwargs: Unpack[Fruit]) -> str: ...
+def foo(**kwargs: Any) -> Any:
+    ...
+
+reveal_type(foo(sort="test", taste=999))  # N: Revealed type is "builtins.str"
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsJoin]
+from typing_extensions import Unpack, TypedDict
+
+class Person(TypedDict):
+    name: str
+    age: int
+
+def foo(*, name: str, age: int) -> None: ...
+def bar(**kwargs: Unpack[Person]) -> None: ...
+
+reveal_type([foo, bar])  # N: Revealed type is "builtins.list[def (*, name: builtins.str, age: builtins.int)]"
+reveal_type([bar, foo])  # N: Revealed type is "builtins.list[def (*, name: builtins.str, age: builtins.int)]"
+[builtins fixtures/dict.pyi]
+
+[case testUnpackKwargsParamSpec]
+from typing import Callable, Any, TypeVar, List
+from typing_extensions import ParamSpec, Unpack, TypedDict
+
+class Person(TypedDict):
+    name: str
+    age: int
+
+P = ParamSpec('P')
+T = TypeVar('T')
+
+def dec(f: Callable[P, T]) -> Callable[P, List[T]]: ...
+
+@dec
+def g(**kwargs: Unpack[Person]) -> int: ...
+
+reveal_type(g)  # N: Revealed type is "def (*, name: builtins.str, age: builtins.int) -> builtins.list[builtins.int]"
+[builtins fixtures/dict.pyi]
```

Comment:
There are some other cases that I think is worth covering:
1. When `TypedDict` has `kwargs` argument (because it is special cased in the implementation)
2. When `TypedDict` has invalid chars for aguments, like `T = TypedDict("T", {"@": int})`
3. When `TypedDict` is empty, no keys
4. When `TypedDict` has a key that overrides other arguments like `T = TypedDict("T", {"a": int})` and `def some(a: str, **kwargs: Unpack[T]): ...` (and similar cases like `def some(a: str, /, **kwargs: Unpack[T])` and `def some(*, a: str, **kwargs: Unpack[T])`)
5. We can also check overrides with compatible / incompatible typed dics

Reply:
I added the tests (and also fixed error messages for incompatible overrides). Btw on non-identifier keys, Python allows them:
```
>>> def foo(**a):
...     print(a)
... 
>>> foo(**{"@": 1})
{'@': 1}
```
Note I already had a test for named argument overlap, so I added the same one for positional-only (which should be allowed).